Userland/DevTools/Profiler/Profile.cpp

```diff
@@ -262,13 +223,72 @@ Result<NonnullOwnPtr<Profile>, String> Profile::load_from_perfcore_file(const St
 
         event.timestamp = perf_event.get("timestamp").to_number<u64>();
         event.type = perf_event.get("type").to_string();
+        event.pid = perf_event.get("pid").to_i32();
         event.tid = perf_event.get("tid").to_i32();
 
         if (event.type == "malloc") {
             event.ptr = perf_event.get("ptr").to_number<FlatPtr>();
             event.size = perf_event.get("size").to_number<size_t>();
         } else if (event.type == "free") {
             event.ptr = perf_event.get("ptr").to_number<FlatPtr>();
+        } else if (event.type == "mmap") {
+            event.ptr = perf_event.get("ptr").to_number<FlatPtr>();
+            event.size = perf_event.get("size").to_number<size_t>();
+            event.name = perf_event.get("name").to_string();
+
+            auto it = current_processes.find(event.pid);
+            if (it != current_processes.end())
+                it->value->library_metadata.handle_mmap(event.ptr, event.size, event.name);
+            continue;
+        } else if (event.type == "munmap") {
+            event.ptr = perf_event.get("ptr").to_number<FlatPtr>();
+            event.size = perf_event.get("size").to_number<size_t>();
+            continue;
+        } else if (event.type == "process_create") {
+            event.parent_pid = perf_event.get("parent_pid").to_number<FlatPtr>();
+            event.executable = perf_event.get("executable").to_string();
+
+            auto sampled_process = adopt_own(*new Process {
+                .pid = event.pid,
+                .executable = event.executable,
+                .start_valid = event.timestamp });
```

Comment:
In situations like this, I usually like to put a `,` after the last member initializer to trick `clang-format` into starting a new line:

```cpp
auto sampled_process = adopt_own(*new Process {
    .pid = event.pid,
    .executable = event.executable,
    .start_valid = event.timestamp,
});
```